src/transformStateWithClones.js

```diff
@@ -6,8 +6,37 @@
  *
  * @return {Object[]}
  */
+
+const loopRemove = (obj, state) => {
+  for (const key of obj) {
+    delete state[key];
+  }
+};
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const prevState = { ...state };
+
+  for (const val of actions) {
+    switch (val.type) {
+      case 'addProperties':
+        Object.assign(prevState, val.extraData);
+        result.push({ ...prevState });
```

Comment:
You can push once after the switch case to avoid repeating code

Reply:
Yes, you are right, thank you!